rpc-protocol/src/main/java/org/sonarsource/sonarlint/core/rpc/protocol/backend/initialize/FeatureFlagsDto.java

```diff
@@ -19,47 +19,54 @@
  */
 package org.sonarsource.sonarlint.core.rpc.protocol.backend.initialize;
 
+import java.util.Collection;
+import java.util.EnumSet;
+import java.util.Set;
+
 /**
  * Optional features toggles of the backend. To ease transition or to accommodate different client needs.
  */
 public class FeatureFlagsDto {
 
-  private final boolean shouldManageSmartNotifications;
-  @Deprecated
-  // not controllable anymore, it is the backend's responsibility to decide
-  private final boolean taintVulnerabilitiesEnabled;
-  private final boolean shouldSynchronizeProjects;
-  private final boolean shouldManageLocalServer;
-  private final boolean enableSecurityHotspots;
-  private final boolean shouldManageServerSentEvents;
-  private final boolean enableDataflowBugDetection;
-  private final boolean shouldManageFullSynchronization;
-  private final boolean enableTelemetry;
-  private final boolean canOpenFixSuggestion;
-  private final boolean enableMonitoring;
+  private final Set<FeatureFlag> featureFlags = EnumSet.noneOf(FeatureFlag.class);
+
+  public FeatureFlagsDto(Collection<FeatureFlag> featureFlags) {
```

Comment:
I think we could accept a Set here?

Reply:
Also related to my comment above, I wonder if we should keep the type FeatureFlags. It's no more flags we represent. What about replacing `FeatureFlagsDto` in `InitializeParams` with the `Set<BackendFeature>`, and deprecate this class?